File: android/src/main/java/com/capacitorjs/plugins/googlemaps/GoogleMapZoomOptions.kt
package com.capacitorjs.plugins.googlemaps

import org.json.JSONObject

class GoogleMapZoomOptions(fromJSONObject: JSONObject) {
    var minZoom: Double? = null
    var maxZoom: Double? = null
    init {
        if (fromJSONObject.has("minZoom")) {
             minZoom = fromJSONObject.getDouble("minZoom")
        }
        if (fromJSONObject.has("maxZoom")) {
            maxZoom = fromJSONObject.getDouble("maxZoom")
        }
    }
}

File: ios/Plugin/GoogleMapZoomOptions.swift
import Foundation
import Capacitor

public struct GoogleMapZoomOptions{
    let minZoom: Float?
    let maxZoom: Float?
    init(fromJSObject: JSObject) throws  {
        minZoom = fromJSObject["minZoom"] as? Float
        maxZoom = fromJSObject["maxZoom"] as? Float
    }
    init(minZoom: Float? = nil, maxZoom: Float? = nil) {
        self.minZoom = minZoom
        self.maxZoom = maxZoom
    }
}
